ext/web/lib.rs

```diff
@@ -314,46 +316,29 @@ impl Resource for TextDecoderResource {
   }
 }
 
-#[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
-struct EncodeIntoResult {
-  read: usize,
-  written: usize,
-}
-
-#[op]
+#[op(v8)]
 fn op_encoding_encode_into(
-  input: String,
+  scope: &mut v8::HandleScope,
+  input: serde_v8::Value,
   buffer: &mut [u8],
-) -> EncodeIntoResult {
-  // Since `input` is already UTF-8, we can simply find the last UTF-8 code
-  // point boundary from input that fits in `buffer`, and copy the bytes up to
-  // that point.
-  let boundary = if buffer.len() >= input.len() {
-    input.len()
-  } else {
-    let mut boundary = buffer.len();
-
-    // The maximum length of a UTF-8 code point is 4 bytes.
-    for _ in 0..4 {
-      if input.is_char_boundary(boundary) {
-        break;
-      }
-      debug_assert!(boundary > 0);
-      boundary -= 1;
-    }
-
-    debug_assert!(input.is_char_boundary(boundary));
-    boundary
+  out_buf: &mut [u8],
+) {
+  let s = v8::Local::<v8::String>::try_from(input.v8_value).unwrap();
+  assert!(out_buf.len() % 4 == 0);
+  // SAFETY: `out_buf` is guaranteed to be aligned to 4 bytes.
```

Comment:
I don't think that's a safe assumption to make, not if any random arraybuffer can be passed in.

Reply:
> I don't think that's a safe assumption to make, not if any random arraybuffer can be passed in.

For `ArrayBuffer`s, it's valid for `.len() >= 4`; `.len() = 0` is a possible case that does evoke UB.

Reply:
okie, added a stricter check. does that look good?